purchase_advance_payment/models/purchase_order.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (C) 2021 ForgeFlow S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html)
+
+from odoo import api, fields, models
+from odoo.tools import float_compare
+
+
+class PurchaseOrder(models.Model):
+
+    _inherit = "purchase.order"
+
+    account_payment_ids = fields.One2many(
+        "account.payment", "purchase_id", string="Pay purchase advanced", readonly=True
+    )
+    amount_residual = fields.Float(
+        "Residual amount",
+        readonly=True,
+        compute="_compute_purchase_advance_payment",
+        store=True,
+    )
+    payment_line_ids = fields.Many2many(
+        "account.move.line",
+        string="Payment move lines",
+        compute="_compute_purchase_advance_payment",
+        store=True,
+    )
+    advance_payment_status = fields.Selection(
+        selection=[
+            ("not_paid", "Not Paid"),
+            ("paid", "Paid"),
+            ("partial", "Partially Paid"),
+        ],
+        store=True,
+        readonly=True,
+        copy=False,
+        tracking=True,
+        compute="_compute_purchase_advance_payment",
+    )
+
+    @api.depends(
+        "currency_id",
+        "company_id",
+        "amount_total",
+        "account_payment_ids",
+        "account_payment_ids.state",
+        "account_payment_ids.move_id",
```

Comment:
```suggestion
```
covered by next dependencies

Reply:
@i-vyshnevska can you please let me know in which dependencies it will be covered?